src/useConditionalTimeout.js

```diff
@@ -39,11 +37,12 @@ const useConditionalTimeout = (fn, milliseconds, condition, options = defaultOpt
         callback.current();
       }, milliseconds);
     }
+    return () => clear();
   }, [condition, milliseconds]);
 
   // when the condition change, clear the timeout
   useEffect(() => {
-    if (prevCondition && condition !== prevCondition && opts.cancelOnConditionChange) {
+    if (!condition && opts.cancelOnConditionChange) {
```

Comment:
I don't really understand this change, why are you changing this part?

Reply:
There is no need to judge prevCondition, condition only is true or false, when condition change, useEffect will run, and condition is true, preCondition must be false

Reply:
Hi @wenzi0github 
thank you for your answer! 

The reason why I'm pointing this out is because 4 tests are not passing:

https://travis-ci.org/github/beautifulinteractions/beautiful-react-hooks

I think you need to fix them so I can merge the PR

Reply:
1. IntersectionObserver only in window, so we should check if current is client, and I changed global to window in test/useViewportSpy.spec.js;
2. when component is unmounted(`rerender(null)`), setTimeout or setInterval should be cleared, spy.called will be false, I think;

Reply:
Hi @wenzi0github 
I'm sorry it took me so long to respond you. 

I understand your improvements on the library and I definitely agree with you. 

Please fix the broken tests so I can finally merge your PR :) 

thank you very much for the effort! 

cheers! :)

Reply:
@wenzi0github please be aware I've updated the project dependencies this morning so you would like to rebase